tools/RegisterAndMonitorImport/RegisterAndMonitorImport.cs

```diff
@@ -0,0 +1,456 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.RegisterAndMonitorImport
+{
+    internal static class RegisterAndMonitorImport
+    {
+        private static readonly string TokenEndpoint = ConfigurationManager.AppSettings["TokenEndpoint"] ?? string.Empty;
+        private static readonly string TokenGrantType = ConfigurationManager.AppSettings["grant_type"] ?? string.Empty;
+        private static readonly string TokenClientId = ConfigurationManager.AppSettings["client_id"] ?? string.Empty;
+        private static readonly string TokenClientSecret = ConfigurationManager.AppSettings["client_secret"] ?? string.Empty;
+        private static readonly string TokenResource = ConfigurationManager.AppSettings["FhirEndpoint"] ?? string.Empty;
+        private static readonly string ResourceType = ConfigurationManager.AppSettings["ResourceType"] ?? string.Empty;
+        private static readonly string ContainerName = ConfigurationManager.AppSettings["ContainerName"] ?? string.Empty;
+        private static readonly string ConnectionString = ConfigurationManager.AppSettings["ConnectionString"] ?? string.Empty;
+        private static readonly string FhirEndpoint = TokenResource;
+        private static readonly int NumberOfBlobsForImport = int.Parse(ConfigurationManager.AppSettings["NumberOfBlobsForImport"] ?? "1");
+        private static readonly TimeSpan ImportStatusDelay = TimeSpan.Parse(ConfigurationManager.AppSettings["ImportStatusDelay"]);
```

Comment:
When there are this many properties, could we just bind to a configuration object? I think it would simplify all this a lot